x-pack/plugins/asset_manager/server/lib/accessors/containers/get_containers.ts

```
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import { Asset } from '../../../../common/types_api';
+import { GetContainersOptionsPublic } from '../../../../common/types_client';
+import {
+  AssetClientDependencies,
+  AssetClientOptionsWithInjectedValues,
+} from '../../asset_client_types';
+import { parseEan } from '../../parse_ean';
+import { collectContainers } from '../../collectors';
+import { validateStringDateRange } from '../../validators/validate_date_range';
+
+export type GetContainersOptions = GetContainersOptionsPublic & AssetClientDependencies;
+export type GetContainersOptionsInjected =
+  AssetClientOptionsWithInjectedValues<GetContainersOptions>;
+
+export async function getContainers(
+  options: GetContainersOptionsInjected
+): Promise<{ containers: Asset[] }> {
+  validateStringDateRange(options.from, options.to);
+
+  const metricsIndices = await options.metricsClient.getMetricIndices({
+    savedObjectsClient: options.savedObjectsClient,
+  });
+
+  const filters: QueryDslQueryContainer[] = [];
+
+  if (options.filters?.ean) {
+    const ean = Array.isArray(options.filters.ean) ? options.filters.ean[0] : options.filters.ean;
+    const { kind, id } = parseEan(ean);
+
+    // if EAN filter isn't targeting a container asset, we don't need to do this query
+    if (kind !== 'container') {
+      return {
+        containers: [],
+      };
+    }
```

wondering if this should throw instead, providing an unsupported kind to the function can be considered an error while returning empty array can hide misuse of the api during dev

---

So this if block handles the case where someone looked up a valid EAN that doesn't match any container assets. I think when we want to run this query inside of something like `getAssets()`, this will need to be tolerant of the case that someone is looking for `host:xyz` and just return `[]` for other asset kinds. 

That said, we don't validate that the EAN contains only the valid, allowable `kind` values. Maybe we can use the runtime type's `.is()` method inside of `parseEan()` to do that, what do you think?

---

See: ef64e4a8c870e7a962d3928ba7c4c43bda6fe941